### PR TITLE
Initial files for keycloak support - may have to be tweaked

### PR DESCRIPTION
**Pull request by n1zea144**
Config files for Keycloak service support on k8s.  The current setup is to deploy keycloak via helm charts.  In addition to database config support, the chart provides support for ingress modification (which we take advantage of).

Initially, Keycloak is being setup to satisfy needs of the GENIE project.  As such, Keycloak is being deployed and pointing to the GENIE RDS node.  In the long run, we should decide if we want a general Keycloak service pointing to a general database for use by all password protected cBioPortal instances or if we want individual instances.  Its seems like a single Keycloak instance with proper use of realms should be the way to go.  If this is true, we will want to repoint the Keycloak service to a different RDS node.

In addition to these configurations, a DNS record has been created which maps requests going from keycloak.cbioportal.org to the DNS name for the AWS load balancer.

### CODE REVIEW

File: ingress/ingress_keycloak.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: keycloak-ingress
```
Comment (n1zea144) [inline ingress/ingress_keycloak.yml]: Should this be in a separate namespace?

File: ingress/ingress_keycloak.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    ingress.kubernetes.io/proxy-connect-timeout: "300"
+    ingress.kubernetes.io/proxy-read-timeout: "300"
+    ingress.kubernetes.io/proxy-send-timeout: "300"
+    # ingress.kubernetes.io/large-client-header-buffers: "4 32k"
+    # increae max response size to avoid 413 errors see
+    # https://github.com/kubernetes/ingress-nginx/issues/1824
+    nginx.ingress.kubernetes.io/proxy-body-size: 512m
+    ingress.kubernetes.io/proxy-body-size: 512m
+    # add proxy protocol to header
+    service.beta.kubernetes.io/aws-load-balancer-proxy-protocol: '*'
```
Comment (inodb) [inline ingress/ingress_keycloak.yml]: ☝️ some of this stuff is cBioPortal specific, prolly doesn't hurt to add it, but just fyi

**Comment by inodb** [comment]
Just so I remember: what happens when keycloak goes down? Are the permissions stored in a persistent manner? Do we need backup that drive in some way?

File: keycloak/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Keycloak
+
+Install the truststore as a secret into the k8s cluster.  The truststore contains required certificates for proper SSL connections to AWS-RDS nodes:
+```
+kubectl create secret generic aws-rds-tls-jks --from-file=<TRUSTSTORE_FILE>
+```
+Note: the <TRUSTSTORE> file can be found at [this location](./aws-rds-tls.jks) in the repository.
```
Comment (inodb) [inline keycloak/README.md]: @n1zea144 just curious where does one download this one?

**Comment by n1zea144** [comment]
@inodb Some prior conversations have become outdated due to recent updates to the PR.  More text was added to README.md.  The yaml was also updated to allow for custom login pages.  Currently a custom login page has been added for the GENIE private portal.  A similar pattern will be followed to add a login page for HTAN.

**Comment by n1zea144** [comment]
> Just so I remember: what happens when keycloak goes down? Are the permissions stored in a persistent manner? Do we need to backup that drive in some way?

Most configuration is saved in the MySQL database hosted on the AWS-RDS node.  One exception is custom login pages, which are packaged as themes and delivered to the chart via a docker image.  This is explained further in the README.md file that is part of this PR.

**Pull request merged**